src/main/kotlin/pt/up/fe/ni/website/backend/model/dto/Dto.kt

```diff
@@ -0,0 +1,26 @@
+package pt.up.fe.ni.website.backend.model.dto
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import javax.validation.ConstraintViolationException
+import javax.validation.Validation
+
+open class Dto<T : Any> {
+
+    companion object {
+        private val objectMapper = ObjectMapper()
+        private val validatorFactory = Validation.buildDefaultValidatorFactory()
+    }
+
+    fun update(entity: T): T {
+        val newEntity = objectMapper.updateValue(entity, this)
+
+        val validator = validatorFactory.validator
+        val violations = validator.validate(newEntity)
+
+        if (violations.isNotEmpty()) {
+            throw ConstraintViolationException(violations)
```

Comment:
This exception will not work with current `ErrorController`. Hopefully we only need to update the exception handled there.

ModelMapper uses `ValidationException` so watch out for that if you use it